crates/engine/tree/src/tree/payload_processor/sparse_trie.rs

```diff
@@ -108,7 +108,26 @@ where
     ///
     /// NOTE: This function does not take `self` by value to prevent blocking on [`SparseStateTrie`]
     /// drop.
-    pub(super) fn run(&mut self) -> Result<StateRootComputeOutcome<A>, ParallelStateRootError> {
+    ///
+    /// # Returns
+    ///
+    /// - State root computation outcome.
+    /// - Accounts trie that needs to be cleared and re-used to avoid reallocations.
+    pub(super) fn run(
+        &mut self,
+    ) -> (Result<StateRootComputeOutcome, ParallelStateRootError>, SparseTrie<A>) {
+        // run the main loop to completion
+        let result = self.run_to_completion();
+        // take the account trie so that we can reuse its already allocated data structures.
+        let trie = self.trie.take_accounts_trie();
+
+        (result, trie)
+    }
+
+    /// Run the sparse trie task to completion.
+    ///
+    /// See [`Self::run`] for more information.
+    fn run_to_completion(&mut self) -> Result<StateRootComputeOutcome, ParallelStateRootError> {
```

Comment:
tbh i think only one of these should have `run` in it? `run` and `run_to_completion` seem kinda similar